packages/firestore/src/remote/serializer.ts

```diff
@@ -1008,6 +1009,14 @@ export class JsonProtoSerializer {
   }
 
   toQueryTarget(query: Query): api.QueryTarget {
+    // limitToLast is implemented client-side and must be "normalized" before
+    // sending to the backend.
+    query = query.convertLimitToFirstIfNecessary();
```

Comment:
I think we should remove this (and fix calling code instead) and the assert below should become query.isTargetQuery == true